app/Services/ImportServiceProvider.js

```diff
@@ -0,0 +1,173 @@
+const { Group, VocabularyCard, Translation, Drawer, sequelize } = require('../../database');
+const ApiError = require('../utils/ApiError.js');
+const httpStatus = require('http-status');
+const { createDrawers } = require('./DrawerServiceProvider.js');
+const { createLanguagePackage } = require('./LanguagePackageServiceProvider.js');
+const { drawers } = require('../utils/constants.js');
+
+async function storeGroupVocabulary(
+  { name, description, VocabularyCards },
+  userId,
+  languagePackageId,
+  active,
+  activate
+) {
+  const transaction = await sequelize.transaction();
+
+  try {
+    const group = await Group.create(
```

Comment:
I think we need to check here, if the langPackage exists. We can do that by moving that into a separate try catch (don't forget the rollback) or we add a check before.